src/Tools/dotnet-watch/src/LaunchBrowserFilter.cs

```diff
@@ -117,18 +111,30 @@ private void OnOutput(object sender, DataReceivedEventArgs eventArgs)
                 var process = (Process)sender;
                 process.OutputDataReceived -= OnOutput;
 
-                if (!_browserLaunched)
+                if (!_attemptedBrowserLaunch)
                 {
+                    _attemptedBrowserLaunch = true;
+
                     _reporter.Verbose("Launching browser.");
+
                     try
                     {
                         LaunchBrowser(launchUrl);
-                        _browserLaunched = true;
                     }
                     catch (Exception ex)
                     {
-                        _reporter.Output($"Unable to launch browser: {ex}");
-                        _canLaunchBrowser = false;
+                        _reporter.Verbose($"Unable to launch browser: {ex}");
+                        _browserProcess = null;
+                    }
+
+                    if (_browserProcess is null || _browserProcess.HasExited)
+                    {
+                        // dotnet-watch, by default, relies on URL file association to launch browsers. On Windows and MacOS, this works fairly well
+                        // where URLs are associated with the default browser. On Linux, this is a bit murky.
+                        // From emperical observation, it's noted that failing to launch a browser results in either Process.Start returning a null-value
+                        // or for the process to have immediately exited.
+                        // We can use this to provide a helpful message.
+                        _reporter.Output($"Unable to launch the browser. Navigate to {launchUrl}");
```

Comment:
If there was an exception, won't they get "Unable to launch browser" twice, does that matter.

Reply:
One is with verbose logging enabled, which isn't common and normally filtered out. @pranavkm maybe change the first log though to remove the duplication?

Reply:
@pranavkm does this still need to be addressed?